### ISSUE TITLE
maximum recursion exceed or get stuck

### ISSUE THREAD

**Issue by haruji** [keras-team/keras#7968]
Here I encounter an error with LocallyConnected1D layer using theano backend .
The error information shows 'Maximum recursion exceed'.
But when I enlarge the limit by setrecursionlimit(10000), the program gets stuck over 30 mins and it seems that it will not start to train.

The code is:
x=keras.layers.local.LocallyConnected1D(2,3,strides=1,regularizer=keras.regularizers.l2(wd))(inputs)
x= keras.layers.normalization.BatchNormalization()(x)
x=keras.layers.Activation('relu')(x)

x = keras.layers.core.Flatten()(x)

outputs=keras.layers.core.Dense(1)(x)

If I change local layer into convolution layer, the code can be executed correctly, so I suppose there is somthing wrong with LocallyConnected1D layer.

My theano version is 0.10.0beta2.dev-f6f45e57a0151fa91b04de60b27f01eceaa0c17a
batch_size=256
The number of dimension of inputs is 3000.
